locations/spiders/pick_n_pay.py

```diff
@@ -0,0 +1,49 @@
+import scrapy
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
+
+
+class PickNPaySpider(scrapy.Spider):
+    name = "pick_n_pay"
+    item_attributes = {"brand": "Pick n Pay", "brand_wikidata": "Q7190735"}
+    start_urls = ["https://api.pnp.co.za/stores/v2"]
+
+    def parse(self, response, **kwargs):
+        for store in response.json():
+            store["id"] = store.pop("storeId")
+            address = store["storeAddress"]
+            store["housenumber"] = address.get("streetNo")
+            store["street"] = address.get("street")
+            store["suburb"] = address.get("suburb")
+            store["city"] = address.get("city")
+            province = address.get("province") or {}  # can be JSON null
+            store["province"] = province.get("name")
+            store["country"] = address.get("country")
+            store["postcode"] = address.get("postalCode")
+            item = DictParser.parse(store)
+
+            oh = OpeningHours()
+            for day in store.get("tradingHours", []):
+                # dayId is 1 to 8, 1 being Monday
+                # dayId of 8 is only present on some 24/7 stores so we ignore it
+                if day["dayId"] < 8:
+                    oh.add_range(DAYS[day["dayId"] - 1], day["openTime"], day["closeTime"])
+            item["opening_hours"] = oh
+
+            if store["storeType"] == "CLOTHING":
+                item.update({"brand": "Pick n Pay Clothing", "brand_wikidata": "Q122964352"})
+            elif store["storeType"] == "LIQUOR":
+                item.update({"brand": "Pick n Pay Liquor", "brand_wikidata": "Q122764458"})
+            elif store["storeType"] == "EXPRESS":
+                item.update({"brand": "Pick n Pay Express", "brand_wikidata": "Q122764443"})
+            elif store["storeType"] in ("BOXER", "BOXER PUNCH", "BOXER SUPERSTOR"):
```

Comment:
Boxer Punch is described on page 13 of https://www.picknpayinvestor.co.za/downloads/annual-report/2023/integrated-annual-report-2023-spreads.pdf as being a separately branded small footprint format Boxer Supermarket, and probably should have a separate Wikidata item, even though it's just two stores.

I'm not too bothered by this comment being ignored though, given it is just two stores and hardly much of a separate brand as a result.